app/models/subscription.rb

```diff
@@ -11,6 +11,10 @@ class Subscription < ApplicationRecord
   validates :search_criteria, uniqueness: { scope: %i[email expires_on frequency] }
 
   scope :ongoing, -> { where('expires_on >= current_date') }
+  scope :first_reminder_not_sent, -> { where(first_reminder_sent: false) }
+  scope :final_reminder_not_sent, -> { where(final_reminder_sent: false) }
+  scope :due_first_expiry_notice, -> { first_reminder_not_sent.where('expires_on <= ?', 1.week.from_now.to_date) }
+  scope :due_final_expiry_notice, -> { final_reminder_not_sent.where('expires_on <= ?', 1.day.from_now.to_date) }
```

Comment:
Maybe we should exclude final expiries from first expiries? That is, users should only get one email regarding expiry on a given day. Though this should be an edge case, so happy to continue as is.

Reply:
TODO: Clarify if this is checking the penultimate day or the final day.